@kgerheiser can you give me some CMake help with this one?

The test compile is failing like this:

```
[ 77%] Linking Fortran executable test_g2 /usr/bin/ld: cannot find
-lw3emc:w3emc_d: No such file or directory collect2: error: ld
returned 1 exit status make[2]: ***
[tests/CMakeFiles/test_g2.dir/build.make:105: tests/test_g2] Error 1
make[1]: *** [CMakeFiles/Makefile2:1127:
tests/CMakeFiles/test_g2.dir/all] Error 2 make: *** [Makefile:146:
all] Error 2
```

Yet in my src/CMakeLists.txt file I have this, which I thought would work:

`  target_link_libraries(${lib_name} PUBLIC bacio::bacio ${PNG_LIBRARIES} ${JASPER_LIBRARIES} w3emc:w3emc_${kind})
`
What am I doing wrong?

--- a/.github/workflows/Intel.yml
+++ b/.github/workflows/Intel.yml
@@ -47,6 +47,21 @@ jobs:
         cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
         make -j2
         make install
+    - name: checkout-w3emc
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-w3emc
+        path: w3emc
+        ref: develop
+
+    - name: build-w3emc
+      run: |
+        cd w3emc
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_PREFIX_PATH=~/bacio -DCMAKE_INSTALL_PREFIX=~/w3emc
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -80,7 +95,7 @@ jobs:
         cd g2
         mkdir build 
         cd build
-        cmake .. -DOPENMP=ON -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio
+        cmake .. -DOPENMP=ON -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH="~/bacio;~/w3emc"
         make -j2 VERBOSE=1
     
     - name: test

--- a/.github/workflows/gcc_docs_test-coverage.yml
+++ b/.github/workflows/gcc_docs_test-coverage.yml
@@ -32,6 +32,21 @@ jobs:
         cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
         make -j2
         make install
+    - name: checkout-w3emc
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-w3emc
+        path: w3emc
+        ref: develop
+
+    - name: build-w3emc
+      run: |
+        cd w3emc
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_PREFIX_PATH=~/bacio -DCMAKE_INSTALL_PREFIX=~/w3emc
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -67,7 +82,7 @@ jobs:
         mkdir build
         doxygen --version
         cd build
-        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -Wall" -DCMAKE_C_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -Wall"
+        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH="~/bacio;~/w3emc" -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -Wall" -DCMAKE_C_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -Wall"
         make -j2 VERBOSE=1 
 
     - name: test

--- a/.github/workflows/gnu_bacio-2.4.1.yml
+++ b/.github/workflows/gnu_bacio-2.4.1.yml
@@ -36,6 +36,21 @@ jobs:
         cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
         make -j2
         make install
+    - name: checkout-w3emc
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-w3emc
+        path: w3emc
+        ref: develop
+
+    - name: build-w3emc
+      run: |
+        cd w3emc
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_PREFIX_PATH=~/bacio -DCMAKE_INSTALL_PREFIX=~/w3emc
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -72,7 +87,7 @@ jobs:
         doxygen --version
         ls ~/bacio
         cd build
-        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0"
+        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH="~/bacio;~/w3emc" -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0"
         make -j2 VERBOSE=1 
 
     - name: test

--- a/.github/workflows/macOS.yml
+++ b/.github/workflows/macOS.yml
@@ -31,6 +31,21 @@ jobs:
         cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
         make -j2
         make install
+    - name: checkout-w3emc
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-w3emc
+        path: w3emc
+        ref: develop
+
+    - name: build-w3emc
+      run: |
+        cd w3emc
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_PREFIX_PATH=~/bacio -DCMAKE_INSTALL_PREFIX=~/w3emc
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -64,7 +79,7 @@ jobs:
         cd g2
         mkdir build
         cd build
-        cmake -DCMAKE_PREFIX_PATH="~/Jasper;~/bacio" ..
+        cmake -DCMAKE_PREFIX_PATH="~/Jasper;~/bacio;~/w3emc" ..
         make -j2
 
     - name: test

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -77,7 +77,7 @@ foreach(kind ${kinds})
 
   add_library(${lib_name} STATIC $<TARGET_OBJECTS:${lib_name}_f> $<TARGET_OBJECTS:${lib_name}_c>)
 
-  target_link_libraries(${lib_name} PUBLIC bacio::bacio ${PNG_LIBRARIES} ${JASPER_LIBRARIES} w3emc:w3emc_${kind})
+  target_link_libraries(${lib_name} PUBLIC bacio::bacio ${PNG_LIBRARIES} ${JASPER_LIBRARIES} w3emc::w3emc_${kind})
 
   target_include_directories(${lib_name} INTERFACE
     $<BUILD_INTERFACE:${module_dir}>

